File: src/IO/dislin-11.0/python/areaf.py
#!/usr/bin/env python
import dislin

ix  = [0, 300, 300,   0]
iy  = [0,   0, 400, 400]
ixp = [0, 0, 0, 0]
iyp = [0, 0, 0, 0]

ctit = 'Shading Patterns (AREAF)'

dislin.metafl ('cons')
dislin.disini ()
dislin.setvlt ('small')
dislin.pagera ()
dislin.complx ()

dislin.height (50)
nl = dislin.nlmess (ctit)
dislin.messag (ctit, (2970 - nl)/2, 200)

nx0 = 335
ny0 = 350

iclr = 0
for i in range (0, 3):
  ny = ny0 + i * 600
  for j in range (0, 6):
    nx = nx0 + j * 400
    ii = i * 6 + j
    dislin.shdpat (ii)
    iclr = iclr + 1
    dislin.setclr (iclr)
    for k in range (0, 4):
      ixp[k] = ix[k] + nx
      iyp[k] = iy[k] + ny

    dislin.areaf (ixp, iyp, 4)
    nl = dislin.nlnumb (ii, -1)
    nx = nx + (300 - nl) / 2
    dislin.color ('foreground')
    dislin.number (ii, -1, nx, ny + 460) 

dislin.disfin ()

File: src/IO/dislin-11.0/java/areaf.java
import de.dislin.Dislin;

public class areaf {
  public static void main (String args []) {
    int i, j, k, ii, nl, nx, ny, nx0 = 335, ny0 = 350, iclr = 0;
    int ix  [] = {0, 300, 300,   0};
    int iy  [] = {0,   0, 400, 400};
    int ixp [] = {0, 0, 0, 0};
    int iyp [] = {0, 0, 0, 0};
    String ctit = "Shading Patterns (AREAF)";

    Dislin.metafl ("cons");
    Dislin.disini ();
    Dislin.setvlt ("small");
    Dislin.pagera ();
    Dislin.complx ();

    Dislin.height (50);
    nl = Dislin.nlmess (ctit);
    Dislin.messag (ctit, (2970 - nl)/2, 200);

    for (i = 0; i < 3; i++) {
      ny = ny0 + i * 600;
      for (j = 0; j < 6; j++) {
        nx = nx0 + j * 400;
        ii = i * 6 + j;
        Dislin.shdpat (ii);
        iclr++;
        iclr = iclr % 8;
        if (iclr == 0)
           iclr = 8;
        Dislin.setclr (iclr);
        for (k = 0; k < 4; k++) {
          ixp[k] = ix[k] + nx;
          iyp[k] = iy[k] + ny;
        }

        Dislin.areaf (ixp, iyp, 4);
        nl = Dislin.nlnumb (ii, -1);
        nx = nx + (300 - nl) / 2;
        Dislin.color ("foreground");
        Dislin.number (ii, -1, nx, ny + 460); 
      }
    }
    Dislin.disfin ();
  }
}
